org.eclipse.m2e.jdt.tests/src/org/eclipse/m2e/jdt/tests/UnitTestLaunchConfigConfigurationTest.java

```diff
@@ -279,6 +279,33 @@ public void test_configuration_must_be_updated_with_failSafe_config_when_created
 		assertTrue(argLine.contains("-DfailsafeProp1=failsafeProp1Value"));
 	}
 
+	@Test
+	public void test_deferred_variable_are_resolved() throws CoreException, IOException, InterruptedException {
+		// Get launch type
+		ILaunchConfigurationType type = LAUNCH_MANAGER.getLaunchConfigurationType(testType);
+
+		assumeTrue(testType + " support not available", type != null);
+
+		File pomFile = getTestFile("deferredVariables/pom.xml");
+
+		IProject project = importProject(pomFile.getAbsolutePath());
+
+		// create basic unit test
+		createDefaultTest(project, type, "test.SomeTest");
+
+		updateProject(project);
+		waitForJobsToComplete();
+
+		ILaunchConfiguration[] updatedConfigurations = LAUNCH_MANAGER.getLaunchConfigurations(type);
+		assertTrue(updatedConfigurations.length == 1);
+
+		ILaunchConfiguration config = updatedConfigurations[0];
+		String argLine = config.getAttribute(UnitTestSupport.LAUNCH_CONFIG_VM_ARGUMENTS, "");
+		assertTrue(argLine.contains("-javaagent")); // resolved jacoco agent
+		assertTrue(argLine.contains("@{titi.tata}")); // unresolved property, maybe removing the placeholder would be
```

Comment:
I'd say yes - removing it would probably be better ;) 

Although I believe that in a CLI build, it would also not be resolved and stay as is.

Reply:
I hesitated but in the end I preferred a quick failure in the case of an undefined variable than spending time looking for where the problem comes from

@HannesWell, @laeubi, @akurtakov what are your thought on this ?

Reply:
I agree.
First M2E should behave as much as possible as a standalone Maven execution on the CLI.
And second respectively in general a fail-fast behavior is usually better.